downloads/swan-lake-release-notes/swan-lake-preview8/RELEASE_NOTE.md

```diff
@@ -0,0 +1,519 @@
+
+By defining services as participants, all services work as a single unit of work. If any of the services fails, the whole transaction will be reverted and if all the services are successfully called, the transaction will be completed and committed successfully. 
+
+**Defining a Transactional Resource Function of a Service**
+
+```ballerina
+transactional resource function get message(http:Caller caller, http:Request req) {
+    http:Response res = new;
+    callSomeService(res);        
+    callAnotherService(res);
+    checkpanic caller->respond(res);
+}
+```
+
+**Defining  the Transactional Remote Functions of a Client Object**
+
+
+```ballerina
+transactional remote function callMyFirstService() returns @tainted any|error {
+    return self.httpClient->get("/echo/message");
+}
+
+transactional remote function callMySecondService() returns @tainted any|error {
+    return self.httpClient->get("/echo/message");
+}
+```
+
+**Calling the Service**
+
+```ballerina
+transaction {
+    var res1 = client->callMyFirstService();
+   var  res2 = client ->callMySecondService();
+   var x = commit;
+    if (x is error) {
+        // error code
+    } else {
+        // success code
+    }
+}
+```                                                             
+
+#### Standard Library
+
+##### HTTP Module Changes
+
+###### Service Declaration
+
+- Basepath is removed from the `ServiceConfig`. Use the absolute resource path, which begins with `/` as the basePath. The `absolute-resource-path` is optional, which defaults to `/` when not specified.
+- The service type can be added after the `service` keyword as `http:Service`, which is also optional.
+
+**Old Syntax**
+
+```ballerina
+@http:ServiceConfig {
+    basePath: “hello”
+}
+service myService on new http:Listener(9090 {
+
+}
+```
+
+**New Syntax**
+
+```ballerina
+service http:Service /hello on new http:Listener(9090 {
+
+}
+```
+
+###### Resource Function Declaration
+
+- Use the accessor name to specify the HTTP method instead of the `methods` field of the `ResourceConfig` (e.g., `get`).
+- Use `default` the as accessor name when all standard HTTP methods need to be supported (e.g., the passthrough/proxy use case).
+- The resource path segement represents the `path`. The `path` field of the `ResourceConfig` is removed.
+- Use `.` to specify the resource path segement if the path needs to be set as `/`.
+- Path params are specified in the resource path segement within square brackets along with the type. The supported types are string, int, float, boolean, decimal (e.g., `path/[string foo]`).
+- Resource signature params are optional. Even the `Caller` and `Request` are optional and not ordered.
```

Comment:
```suggestion
- Resource function signature parameters are optional. Even the `Caller` and `Request` are optional and not ordered.
```

Reply:
Shall we use "Resource method" instead?

Reply:
Any reason to use `method`? Use designed them as remote/resource functions right

Reply:
Although in the code we write `resource function` and `remote function` they are referred to as resource and remote methods. This is similar to object methods.

Generallly standalone functions are called functions while functions that are bound to something like in an object (functions that are called "on" something) are called methods. Resource and remote methods are special cases of object methods.